### DISCUSSION TITLE
Match with quotes change and replace doesn't work when the cursor is in the match

### DISCUSSION THREAD

**Discussion by otavioschwanck** [helix-editor/helix#10419] [General]
### Summary

The replace should work when cursor is on the match

### Reproduction Steps

Create a simple text:

"helix"

Put the cursor on "

try to replace with mr"[

Also happens with '

### Helix log

.

### Platform

Mac

### Terminal Emulator

Alacritty

### Helix Version

22.12

**Comment by archseer** [comment]
Replace works on selections. `ma'mr'"`

**Comment by otavioschwanck** [comment]
> Replace works on selections. `ma'mr'"`

this input make it happens:

<img width="286" alt="image" src="https://user-images.githubusercontent.com/9551316/212897176-ada8bbd4-85fa-4972-b757-e781914e4959.png">
